pkg/lib/files/files_test.go

```diff
@@ -239,12 +259,89 @@ func TestListDirRecursive(t *testing.T) {
 	require.NoError(t, err)
 	require.ElementsMatch(t, expected, filesListRecursive)
 
-	filesListRecursive, err = ListDirRecursive(tmpDir, true, IgnoreNonPython)
+	filesListRecursive, err = ListDirRecursive(tmpDir, true, emptyExcludes, IgnoreNonPython)
 	expected = []string{
 		filepath.Join("2.py"),
 		filepath.Join("3/1.py"),
 		filepath.Join("3/2/1.py"),
 	}
 	require.NoError(t, err)
 	require.ElementsMatch(t, expected, filesListRecursive)
+
+	filesListRecursive, err = ListDirRecursive(tmpDir, false, excludes)
+	expected = []string{
+		filepath.Join(tmpDir, "2.py"),
+		filepath.Join(tmpDir, "3/2/2.txt"),
+		filepath.Join(tmpDir, "4/2.pyc"),
+		filepath.Join(tmpDir, "4/.git/HEAD"),
+	}
+	require.NoError(t, err)
+	require.ElementsMatch(t, expected, filesListRecursive)
+
+	filesListRecursive, err = ListDirRecursive(tmpDir, true, excludesWithRelativePaths, IgnoreNonPython)
+	expected = []string{
+		filepath.Join("2.py"),
+	}
+	require.NoError(t, err)
+	require.ElementsMatch(t, expected, filesListRecursive)
+}
+
+func TestReadAllIgnorePatterns(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "cortexignore-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	di, err := ReadAllIgnorePatterns(nil)
+	if err != nil {
+		t.Fatalf("Expected not to have error, got %v", err)
+	}
+
+	if diLen := len(di); diLen != 0 {
+		t.Fatalf("Expected to have zero dockerignore entry, got %d", diLen)
+	}
+
+	diName := filepath.Join(tmpDir, ".dockerignore")
+	content := fmt.Sprintf("test1\n/test2\n/a/file/here\n\nlastfile\n# this is a comment\n! /inverted/abs/path\n!\n! \n")
+	err = ioutil.WriteFile(diName, []byte(content), 0777)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	diFd, err := os.Open(diName)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer diFd.Close()
+
+	di, err = ReadAllIgnorePatterns(diFd)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(di) != 7 {
+		t.Fatalf("Expected 5 entries, got %v", len(di))
+	}
+	if di[0] != "test1" {
```

Comment:
You can use `require.Equal(t, di[0], "test1")` here and below if you'd like

Reply:
Agree.